address-space-controller/src/main/java/io/enmasse/controller/AddressFinalizerController.java

```diff
@@ -39,7 +40,17 @@ public Result processFinalizer(final AddressSpace addressSpace) {
 
         ContinuationResult<Address> list = null;
         do {
-            list = addressApi.listAddresses(addressSpaceNamespace, BATCH_SIZE, list, null);
+            try {
+                list = addressApi.listAddresses(addressSpaceNamespace, BATCH_SIZE, list, null);
+            } catch (KubernetesClientException e) {
+                // If not found, the address CRD does not exist so we drop the finalizer
+                if (e.getCode() == 404) {
+                    return Result.completed(addressSpace);
```

Comment:
is there a simple way that we can establish with certainty that the 404 refers to to the absence of the crd (rather than listAddresses returning 404 for some other reason).  I know list returns an empty list of there are none.  My concern is a minor one - and the complication of the additional check may not be worth while.

Reply:
I'm not sure really. One could check if the resource is listed by querying the /apis/enmasse.io/v1beta1 as an additional check, and failing that mark it as finalized. 

Worst case is that addresses would be orphaned.

Reply:
@k-wall Added an attempt to query the API checking if resource exists. Not sure if I like it or not :)

Reply:
Yeah that quite a lot of ugliness..  How about as an alternative simple log a WARN when we go down the 404 branch, that way if we ever go down that path as runtime, they'll be a clue in the log.

Reply:
Ok, I've added a warning and removed the ugliness.